deploy-board/deploy_board/webapp/group_view.py

```diff
@@ -179,6 +180,30 @@ def update_launch_config(request, group_name):
             launchRequest["assignPublicIp"] = True
         else:
             launchRequest["assignPublicIp"] = False
+
+        userPassInSpiffeID = None
+        passInUserDataList = params["userData"].splitlines()
+        for item in passInUserDataList:
+                if ":" in item:
+                    pair = item.split(":")
+                    if pair[0] == "spiffe_id":
+                        userPassInSpiffeID = pair[1]
+
+        group_info = autoscaling_groups_helper.get_group_info(request, group_name)
+        launch_config = group_info.get("launchInfo")
+        if 'userData' in launch_config:
+            userDataList = launch_config['userData'].splitlines()
+            for item in userDataList:
+                if ":" in item:
+                    pair = item.split(":")
+                    if pair[0] == "spiffe_id" and userPassInSpiffeID is None:
+                        log.error("Teletraan does not support user to remove spiffe_id in userData field")
+                        raise TeletraanException("Teletraan does not support user to remove spiffe_id")
+
+                    if pair[0] == "spiffe_id" and pair[1] != userPassInSpiffeID:
+                        log.error("Teletraan does not support user to update spiffe_id in userData field")
+                        raise TeletraanException("Teletraan does not support user to update spiffe_id")
+
```

Comment:
looks like there is still a case that pair[0] != "spiffe_id" and userPassInSpiffeID

Reply:
Nice catch :) Updated, thanks for pointing out.